virl2_client/models/annotation.py

```diff
@@ -365,10 +365,8 @@ def update(
         :param push_to_server: Whether to push the changes to the server.
             Defaults to True; should only be False when used by internal methods.
         """
-        if "type" not in annotation_data:
-            annotation_data["type"] = self._type
-        elif annotation_data["type"] != self._type:
-            raise ValueError("Can't update annotation type.")
+        if annotation_data.get("type") not in (None, self._type):
```

Comment:
@tmikuska incorrect; pipe in python is not 'and', it's the unity operator which merges two dictionaries. I simply made it so `_set_annotation_properties` overrides the `type` parameter no matter what by merging a dictionary that contains it into the original dictionary, overriding the original's `type` key if it has it.

All I effectively did was remove the redundant `if "type" not in annotation_data:             annotation_data["type"] = self._type` from the above function - since it was already done in `_set_annotation_properties` anyway - and then simplified `_set_annotation_properties` to just override the `type`, since there is no situation where we want anything else.

In fact, why do we require `type` to always be present? The API doesn't seem to require it, we always set it to `self._type`, so why not just omit it altogether?

Reply:
I see, I missed the change in _set_annotation_properties...

From what I know "type" is actually the only required property for all annotation endpoints, as you can see in AnnotationBodyUpdate schema for PATCH request. And you cant update annotation to other type.
It is basically an extra check on the client side to prevent sending a request when we know it is meant to do illegal stuff, being type update.